web/src/stream_settings_data.ts

```diff
@@ -50,7 +50,7 @@ function get_subs_for_settings(subs: StreamSubscription[]): SettingsSubscription
     // delegating, so that we can more efficiently compute subscriber counts
     // (in bulk).  If that plan appears to have been aborted, feel free to
     // inline this.
-    return subs.map((sub) => get_sub_for_settings(sub));
+    return subs.filter((sub) => !sub.archived).map((sub) => get_sub_for_settings(sub));
```

Comment:
Why is this change the right place to filter? I would expect to want to do the filtering in higher-level functions closer to where we're actually using them.

Reply:
I think it is the right place to filter. Both functions `get_streams_for_settings_page` and `render_left_panel_superset` depend on `get_subs_for_settings` to get stream related data.
We can  filter streams separately within each function but I don't think its a very good approach.